lib/src/api/model/mod.rs

```diff
@@ -0,0 +1,64 @@
+//! The model module describes public driver and protocol API structures
+//! The reason is that we have internal representations of these objects that contain excessive
+//! information that we do not want to expose or would like to version separately
+use crate::sql::{Object, Uuid, Value};
+use serde::{Deserialize, Serialize};
+use std::fmt::{self, Debug, Display};
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum Action {
+	Create,
+	Update,
+	Delete,
+}
+
+impl Display for Action {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		match *self {
+			Action::Create => write!(f, "CREATE"),
+			Action::Update => write!(f, "UPDATE"),
+			Action::Delete => write!(f, "DELETE"),
+		}
+	}
+}
+
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct Notification {
+	// Live query ID
+	pub id: Uuid,
+	pub action: Action,
+	pub result: Value,
+}
+
+impl Display for Notification {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		let obj: Object = map! {
+			"id".to_string() => self.id.to_string().into(),
```

Comment:
Is there a reason we are converting the UUID to a string instead of keeping it a UUID using `Value::Uuid`?

Reply:
Changed from write object to write fields explicitly